src/helpers/deleteMissingKeys.ts

```diff
@@ -0,0 +1,12 @@
+import { isObject } from './isObject';
+
+export function deleteMissingKeys(target: object, source: object) {
```

Comment:
Please refactor this function to make it immutable.

Reply:
Instead of deleting the keys from `target`, you want to construct a new object that has the keys removed, right? I wonder why though, because it seems to be more complicated to construct a new object (also consider the recursion) than just modify the existing one.